internal/fxlog/spy.go

```diff
@@ -21,64 +21,41 @@
 package fxlog
 
 import (
-	"fmt"
-	"strings"
-
-	"go.uber.org/zap"
+	"reflect"
 )
 
 // Spy is an Fx Printer that captures logged statements. It may be used in
 // tests of Fx logs.
 type Spy struct {
-	entries []Entry
+	events []Event
 }
 
 var _ Logger = &Spy{}
 
-// Log append an Entry.
-func (s *Spy) Log(entry Entry) {
-	s.entries = append(s.entries, entry)
-}
-
-// Messages returns a copy of all captured messages.
-func (s *Spy) Messages() []Entry {
-	msgs := make([]Entry, len(s.entries))
-	copy(msgs, s.entries)
-	return msgs
+// LogEvent appends an Event.
+func (s *Spy) LogEvent(event Event) {
+	s.events = append(s.events, event)
 }
 
-// String returns all logged messages as a single newline-delimited string.
-func (s *Spy) String() string {
-	if len(s.entries) == 0 {
-		return ""
-	}
-	var msg strings.Builder
-	for _, m := range s.entries {
-		fmt.Fprintf(&msg, m.Message)
-		for _, f := range m.Fields {
-			// extra space before f.Key to separate out message.
-			fmt.Fprintf(&msg, "\t%s: %v", f.Key, f.Value)
-		}
-		if m.Stack != "" {
-			fmt.Fprintf(&msg, "\t%q", m.Stack)
-		}
-		msg.WriteString("\n")
-	}
+// Events returns all captured events.
+func (s *Spy) Events() []Event {
+	events := make([]Event, len(s.events))
+	copy(events, s.events)
 
-	return msg.String()
+	return s.events
```

Comment:
Looks like this is returning the original slice and not the copied one:
```suggestion
	return events
```